sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -158,9 +158,13 @@ class CodegenContext {
   /** The variable name of the input row in generated code. */
   final var INPUT_ROW = "i"
 
+  /** The variable name of the input col in generated code. */
+  var INPUT_COLORDINAL = "idx"
+
   /**
     * The map from a variable name to it's next ID.
     */
+  private val freshNameJavaTypes = new mutable.HashMap[String, String]
```

Comment:
It is also used in ```CodeGenerator.getValue()```. This is called from other classes rather than ```DataSourceScan```. Thus, can we keep this in ```CodeGenerator.scala```?